### PR TITLE
Assignment 2

### PR DESCRIPTION
**Pull request by snehaldhadge**

### CODE REVIEW

File: Assignment2_Snehal_Ilawe.sql [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+# Creating View vw_allbatterdata for combining batters_count with local_date and ordering by game 
+# date 
+###################################################################################################
+
+DROP VIEW IF EXISTS vw_allbatterdata;
+
+CREATE VIEW vw_allbatterdata as
+(SELECT hit,atbat,batter,bc.game_id,Date(local_date) as l_dt from batter_counts bc 
+JOIN game g ON g.game_id = bc.game_id 
+ORDER BY game_id ,local_date )
```
Comment (vbihare) [inline Assignment2_Snehal_Ilawe.sql]: Semicolon missing. Hence it interrupts the script.

File: Assignment2_Snehal_Ilawe.sql [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+DROP TABLE IF EXISTS rolling_average;
+
+CREATE TABLE rolling_average  as
+select IFNULL((SELECT SUM(IFNULL(Hit,0))/NULLIF(sum(IFNULL(atbat,0)),0) from vw_allbatterdata vw1 where vw1.l_dt BETWEEN DATE_ADD(vw.l_dt, INTERVAL -100 day)
+AND DATE_SUB(vw.l_dt,INTERVAL 1 DAY) and vw1.batter = vw.batter),0) as Rolling_average , vw.batter,vw.game_id,vw.l_dt 
+from vw_allbatterdata vw 
+ORDER BY vw.game_id;
```
Comment (vbihare) [inline Assignment2_Snehal_Ilawe.sql:63]: This query takes a lot of time to process, is there any way we can reduce the time of computation?

Comment (snehaldhadge) [reply Assignment2_Snehal_Ilawe.sql:63]: The Select query runs within seconds but dont know why when clubbed with CREATE TABLE it is taking really long. Will try to look into options.

Comment (vbihare) [reply Assignment2_Snehal_Ilawe.sql:63]: Okay. I just ran the select query part, because it was taking a lot of time. It looks good to me

**Review by dafrenchyman — APPROVED** [review]
Please check the comments I gave you for the assignment in Canvas. Once you’ve addressed those issues (if any) please merge the PR in!

**Pull request merged**